compiler/rustc_macros/src/newtype.rs

```diff
@@ -137,6 +146,40 @@ impl Parse for Newtype {
             quote! {}
         };
 
+        if ord {
+            derive_paths.push(quote!(Ord));
+            derive_paths.push(quote!(PartialOrd));
+        }
```

Comment:
This is incorrect, but I'm not sure how to get the needed type here.

```
error[E0308]: mismatched types
   --> compiler/rustc_macros/src/newtype.rs:150:31
    |
150 |             derive_paths.push(quote!(Ord));
    |                               ^^^^^^^^^^^ expected struct `syn::Path`, found struct `TokenStream2`
    |
    = note: this error originates in the macro `quote` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Reply:
You can use `parse_quote` instead of `quote` to parse the provided tokens into a `syn` type.